### PR TITLE
Feat/message cloudevents

### PR DESCRIPTION
**Pull request by dominikhorn93**
**Issue**

**Description**

**Checklist**

* Code is easy to understand
* No obvious errors or bugs
* CI/CD Pipelines did run successfully
* Tests were implemented
* Documentation was created

### CODE REVIEW

**Comment by hauptmedia** [comment]
This PR adds infrastructure code for handling CloudEvents. The idea here is to to register a CloudEventsHandler as a Bean for every domain which are then automatically interconnected with a connect-cloudevents module.

**Comment by hauptmedia** [comment]
There are still some things that needs to be sorted out about the architecture, perhaps we can talk about it in our weekly on wednesday

File: connect/connect-cloudevents/cloudevents-kafka/src/main/java/io/miragon/miranum/connect/cloudevents/CloudEventConsumer.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+public class CloudEventConsumer {
+    public static void main(String[] args) {
+        Properties props = new Properties();
+
+        // Other config props
+        props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+        props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, CloudEventDeserializer.class);
+
+        try (KafkaConsumer<String, CloudEvent> consumer = new KafkaConsumer<>(props)) {
```
Comment (DaAnda97) [inline connect/connect-cloudevents/cloudevents-kafka/src/main/java/io/miragon/miranum/connect/cloudevents/CloudEventConsumer.java:20]: :+1: very good. Using try with resources is best practice

File: connect/connect-cloudevents/cloudevents-kafka/src/main/java/io/miragon/miranum/connect/cloudevents/CloudEventConsumer.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+            ConsumerRecords<String, CloudEvent> records = consumer.poll(Duration.ofSeconds(10));
+
+            records.forEach(rec -> {
+                System.out.println(rec.value().toString());
```
Comment (DaAnda97) [inline connect/connect-cloudevents/cloudevents-kafka/src/main/java/io/miragon/miranum/connect/cloudevents/CloudEventConsumer.java:25]: That's where we call our domain services right?

File: connect/connect-cloudevents/cloudevents-kafka/src/main/java/io/miragon/miranum/connect/cloudevents/CloudEventsProducer.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, CloudEventSerializer.class);
+
+        try (KafkaProducer<String, CloudEvent> producer = new KafkaProducer<>(props)) {
+
+            // Build an event
+            CloudEvent event = CloudEventBuilder.v1()
+                    .withId("hello")
+                    .withType("example.kafka")
+                    .withSource(URI.create("http://localhost"))
```
Comment (DaAnda97) [inline connect/connect-cloudevents/cloudevents-kafka/src/main/java/io/miragon/miranum/connect/cloudevents/CloudEventsProducer.java:28]: you could set these props via String[] args

File: examples/gateway-example/gateway-c7/src/main/java/io/miragon/miranum/gateway/c7/GatewayC7Application.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+public class GatewayC7Application {
+
+    public static void main(final String[] args) {
+        SpringApplication.run(GatewayC7Application.class, args);
+    }
+
+    @Bean
+    public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
+        return args -> {
+            System.out.println("Registered Beans: ");
```
Comment (DaAnda97) [inline examples/gateway-example/gateway-c7/src/main/java/io/miragon/miranum/gateway/c7/GatewayC7Application.java:24]: You could use `System.out.printf()` for that

**Review by DaAnda97 — COMMENTED** [review]
Finished the code review. Will run it locally and approve it afterwards :)

**Review by DaAnda97 — APPROVED** [review]
Tested locally and added Readme containing example request

**Pull request merged**